### PR TITLE
Added cookie consent, profile toolbox and updated layout.

### PR DESCRIPTION
**Pull request by aartek**

### CODE REVIEW

File: gruntfile.js [hunk 1]
```diff
@@ -70,6 +70,13 @@ module.exports = function (grunt) {
         }
     };
 
+    var getCustomerConsentManagerUrl = function (regionCode) {
+        if(regionCode.toUpperCase === 'EU'){
```
Comment (golojg) [inline gruntfile.js]: Shouldn't it support stage env? Of course urls shouldn't be hardcoded

File: gruntfile.js [hunk 2]
```diff
@@ -70,6 +70,13 @@ module.exports = function (grunt) {
...
+            return 'https://customer-manager.eu-central.stage.modules.yaas.io'
```
Comment (golojg) [inline gruntfile.js]: wrong url:
 'https://customer-manager.eu-central.modules.yaas.io'

File: gruntfile.js [hunk 3]
```diff
@@ -262,7 +262,6 @@ module.exports = function (grunt) {
 
     //---Specialized-Build-Behaviors--------------------------------------------------------
     grunt.registerTask('singleProjectTask', [
-        'jshint',
```
Comment (golojg) [inline gruntfile.js]: why did you remove it?

File: public/js/app/shared/app-config.js [hunk 4]
```diff
@@ -35,7 +35,7 @@ angular.module('ds.appconfig', [])
                 tenantId = window.location.pathname.substring( 1, pathLength-1 );
             } else {
                 // Dynamic ProjectId is configured and replaced by build script, see gruntfile.
-                tenantId = /*StartProjectId*/ '' /*EndProjectId*/;
+                tenantId = /*StartProjectId*/ 'storksfront' /*EndProjectId*/;
```
Comment (golojg) [inline public/js/app/shared/app-config.js]: empty value should be here

File: public/js/app/shared/app-config.js [hunk 5]
```diff
@@ -48,7 +48,13 @@ angular.module('ds.appconfig', [])
         redirectURI: function() {
             // Dynamic RedirectURI is configured and replaced by build script, see gruntfile.
             return /*StartRedirectURI*/ 'http://example.com' /*EndRedirectURI*/;
-        }
+        },
 
+        builderURL: function () {
+            return  /*StartBuilderUrl*/ 'https://builder.yaas.io/' /*EndBuilderUrl*/;
```
Comment (golojg) [inline public/js/app/shared/app-config.js]: empty value should be here

File: public/js/app/shared/app-config.js [hunk 6]
```diff
@@ -48,7 +48,13 @@ angular.module('ds.appconfig', [])
...
+        },
 
+        consentManagerURL: function () {
+            return  /*StartConsentManagerUrl*/ 'https://customer-manager.us-east.modules.yaas.io/' /*EndConsentManagerUrl*/
```
Comment (golojg) [inline public/js/app/shared/app-config.js]: empty value should be here

File: public/js/app/shared/directives/y-tracking.js [hunk 7]
```diff
@@ -89,22 +169,44 @@ angular.module('ds.ytracking', [])
             };
         }])
     .factory('ytrackingSvc', ['yTrackingLocalStorageKey', '$http', 'localStorage', '$window', '$timeout', 'GlobalData', 'settings', 'appConfig', 'CookieSvc',
-        function (yTrackingLocalStorageKey, $http, localStorage, $window, $timeout, GlobalData, settings, appConfig, CookieSvc) {
+        function (yTrackingLocalStorageKey, $http, localStorage, $window, $timeout, GlobalData, settings, appConfig, CookieSvc, $q) {
```
Comment (golojg) [inline public/js/app/shared/directives/y-tracking.js]: please remove $q

File: public/js/app/shared/services/cookie-svc.js [hunk 8]
```diff
@@ -80,6 +87,20 @@ angular.module('ds.shared')
                     return consentReferenceCookie.consentReference;
                 }
                 return consentReferenceCookie;
+            },
+
+            setConsentReferenceTokenCookie: function (consentReferenceToken, expiresIn) {
+                ipCookie.remove(settings.consentReferenceTokenCookie);
+                var consentReferenceTokenCookie = new ConsentReferenceTokenCookie(consentReferenceToken);
+                ipCookie(settings.consentReferenceTokenCookie, JSON.stringify(consentReferenceTokenCookie), { expirationUnit: 'seconds', expires: expiresIn ? expiresIn : defaultExpirySeconds });
```
Comment (golojg) [inline public/js/app/shared/services/cookie-svc.js:95]: expiresIn is not set in y-tracking.js so defaultExpirySeconds is used, but defaultExpirySeconds is set to ~28h. That cookie should live longer.
Please also change setConsentReferenceCookie()

File: public/js/app/shared/directives/y-tracking.js [hunk 9]
```diff
@@ -149,37 +251,25 @@ angular.module('ds.ytracking', [])
             };
 
             /**
-            * Function that process piwik requests
-            */
+             * Function that process piwik requests
+             */
             var processPiwikRequest = function (e) {
```
Comment (golojg) [inline public/js/app/shared/directives/y-tracking.js]:
```
  var processPiwikRequest = function (e) {
                if (isGranted() && getConsentReference()) {
                    //Get object from query parameters
                    var obj = getPiwikQueryParameters(e);
                    //Make post request to service
                    makePiwikRequest(obj);
                }
              };
```

**Review by golojg — CHANGES_REQUESTED** [review]

File: gruntfile.js (hunk 1 above)
Comment (aartek) [reply gruntfile.js]: Done. --region=stage

File: gruntfile.js (hunk 2 above)
Comment (aartek) [reply gruntfile.js]: fixed

File: gruntfile.js (hunk 3 above)
Comment (aartek) [reply gruntfile.js]: restored

**Review by golojg — APPROVED** [review]
